src/vs/workbench/parts/extensions/electron-browser/extensionsWorkbenchService.ts

```diff
@@ -108,7 +108,7 @@ class Extension implements IExtension {
 			return this.local.changelogUrl;
 		}
 
-		return ''; // Hopefully we will change this once the gallery will support that.
+		return this.gallery && this.gallery.assets.changelog;
```

Comment:
Actually no...

Make those two getters `private` and use them in the `getReadme` and `getChangelog` methods to get a hold of the URL.